packages/babel-generator/src/generators/statements.ts

```diff
@@ -303,17 +311,27 @@ export function VariableDeclaration(
   //       bar = "foo";
   //
 
-  let separator;
+  let separator: PrintJoinOptions["separator"] | undefined;
   if (hasInits) {
     separator =
-      node.kind === "const"
-        ? constDeclarationIndent
+      kind === "const" || kind === "using"
+        ? constOrUsingDeclarationIndent
         : variableDeclarationIndent;
   }
 
   //
 
-  this.printList(node.declarations, node, { separator });
+  let iterator: PrintJoinOptions["iterator"] | undefined;
+  if (kind === "using") {
+    // Ensure no line break between `using` and the first declarator
+    iterator = (_, i: number) => {
+      if (i === 0) {
+        this._noLineTerminator = _noLineTerminator;
```

Comment:
Shouldn't we also unset `_noLineTerminator` after printing?

Reply:
Do you mean unset `_noLineTerminator` after printing the `using` keyword? The variable `_noLineTerminator` here stores the value of `this.noLineTerminator` before we print `using`, and then we restore `this._noLineTerminator` after the first declarator is printed.